### PR TITLE
feat: adding documentation with vitepress

### PR DESCRIPTION
**Pull request by JessicaSachs**
Hello! This pull request adds infrastructure for a Vitepress website. The documentation is largely copy/pasted from @jeffbeltran 's "Un-official faker.js" docs [here](https://github.com/JeffBeltran/faker.js-docs).

Should address #22 and then lay the foundation for building #46 (via the /playground/ page in `docs/playground/index.md`).

There is a `netlify.toml` file to deploy on Netlify and we have a free account sponsored by Netlify that will be setup tomorrow morning. Here's a preview of the site on my personal Netlify account https://61dd4313b4122d0007e9a5c4--boring-colden-7dfbbb.netlify.app/

I will say that there is a LOT of documentation that needs to be updated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Faker
+# faker.js
```
Comment (Shinigami92) [inline README.md:1]: I think we explicitly wanted to _rebrand_ it as `Faker`

File: docs/.vitepress/theme/components/Badge.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<script setup lang="ts">
+import { computed } from 'vue'
```
Comment (Shinigami92) [inline docs/.vitepress/theme/components/Badge.vue:2]: Seems this import is not needed :thinking:

File: docs/.vitepress/theme/components/Playground.vue [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<script setup lang="ts">
+import { ref, computed } from 'vue'
```
Comment (Shinigami92) [inline docs/.vitepress/theme/components/Playground.vue:2]: `computed` also not needed here :thinking:

File: docs/api/address.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+Generates a random localized city name. The format string can contain any method provided by faker wrapped in `{{}}`, e.g. `{{name.firstName}}` in order to build the city name.
+
+::: tip
+If no format string is provided one of the following is randomly used:
+
+- `{{address.cityPrefix}} {{name.firstName}}{{address.citySuffix}}`
+- `{{address.cityPrefix}} {{name.firstName}}`
+- `{{name.firstName}}{{address.citySuffix}}`
+- `{{name.lastName}}{{address.citySuffix}}`
+  :::
```
Comment (Shinigami92) [inline docs/api/address.md:17]: Offset may be wrong

File: docs/api/address.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+```js
+faker.address.city() // Lake Raoulfort
+faker.address.city({{name.lastName}}{{address.citySuffix}}) // Powlowski port
```
Comment (Shinigami92) [inline docs/api/address.md:21]: I assume this needs quotes :thinking:

File: docs/guide/recent-faqs.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+**Yarn**
+```shell
+yarn add faker@5.5.3 -D
+```
+
+**pnpm**
+```shell
+pnpm add -D faker@5.5.3
```
Comment (Shinigami92) [inline docs/guide/recent-faqs.md:50]: Needs updated package name

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,72 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
```
Comment (Shinigami92) [inline tsconfig.json:3]: Please remove all the comments
Let's use a simple config
It also could conflict with the config coming in #72 :thinking: Is it possible to just omit this file for now?

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
Could you run prettier over the docs?

File: README.md (hunk 1 above)
Comment (JessicaSachs) [reply README.md:1]: OK yeah I wasn't sure because it was so inconsistent and then the logo looked a little weird. I agree a re-brand is a good idea.

File: docs/.vitepress/theme/components/Badge.vue (hunk 2 above)
Comment (JessicaSachs) [reply docs/.vitepress/theme/components/Badge.vue:2]: Yeah, we really need a linter.

File: tsconfig.json (hunk 7 above)
Comment (JessicaSachs) [reply tsconfig.json:3]: I'm not sure. I agree that it may conflict. I'm happy to use mjs/js instead of ts and let the real typescript PR come in.

File: docs/api/address.md (hunk 5 above)
Comment (JessicaSachs) [reply docs/api/address.md:21]: Yeah, I largely copy/pasted this in from the existing docs.

File: docs/.vitepress/theme/components/Badge.vue (hunk 2 above)
Comment (Shinigami92) [reply docs/.vitepress/theme/components/Badge.vue:2]: We have eslint, but not fully setup yet
Prio for me: TS migration

File: tsconfig.json (hunk 7 above)
Comment (JessicaSachs) [reply tsconfig.json]: Yeah I had no problems swapping to mjs.

**Comment by prisis** [comment]
Can you update the year from 2020 to 2022 :D

File: package-lock.json [hunk 8]
```diff
@@ -1,14921 +1,162 @@
 {
   "name": "faker",
   "version": "5.5.3",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (Shinigami92) [inline package-lock.json]: Please recreate the lock file with node 16, npm 8

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by prisis — APPROVED** [review]

File: package.json [hunk 9]
```diff
@@ -7,15 +7,19 @@
...
     "jsdoc": "./node_modules/.bin/gulp jsdoc",
     "format": "prettier --write .",
     "lint": "node_modules/.bin/eslint ./lib/animal.js",
     "test": "node_modules/.bin/mocha test/*.*.js",
-    "coverage": "nyc report --reporter=text-lcov | coveralls"
+    "coverage": "nyc report --reporter=text-lcov | coveralls",
+    "docs:dev": "vitepress dev docs",
+    "docs:build": "vitepress build docs",
+    "docs:serve": "vitepress serve docs",
+    "postinstall": "npm run browser"
```
Comment (Shinigami92) [inline package.json:22]: Not sure if we want this right now
I reject currently some PRs cause they update these files and we cannot review them cause they are to huge
I don't trust folks in that and we should generate them before release or so on our side

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: package.json (hunk 9 above)
Comment (damienwebdev) [inline package.json:22]: What is the purpose of this `postinstall`?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: package.json (hunk 9 above)
Comment (JessicaSachs) [reply package.json:22]: When building the docs + playground, we need to load faker into the browser. Until we have a way to require faker at runtime we need to require the browserified dist.

TLDR: 

Running `npm run docs:dev` and navigating to `localhost:3000/playground/` should not crash.

Comment (JessicaSachs) [reply package.json:22]: This PR needs to go out.

Comment (Shinigami92) [reply package.json:22]: Maybe we could do `npm run browser && vitepress build docs` for now or something like this
I just wont like to do it after install, cause currently (like I said) it's not reviewable and insecure

Comment (damienwebdev) [reply package.json:22]: Can we instead run this `npm run browser` as a part of `docs:dev` then?

```
npm run browser && vitepress dev docs
```

**Review by Shinigami92 — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
lgtm
